Build failure: gvisor

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

add pkgs.gvisor to `virtualisation.podman.extraPackages`

or probably install it just like that

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

https://hydra.nixos.org/build/324791494

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/cykfn3i15bl9phy07ya00k2cbx9g2cgj-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage ./runsc
# gvisor.dev/gvisor/pkg/sync
pkg/sync/runtime_constants_go126.go:22:2: WaitReasonSelect redeclared in this block
        pkg/sync/runtime_constants_go125.go:22:2: other declaration of WaitReasonSelect
pkg/sync/runtime_constants_go126.go:23:2: WaitReasonChanReceive redeclared in this block
        pkg/sync/runtime_constants_go125.go:23:2: other declaration of WaitReasonChanReceive
pkg/sync/runtime_constants_go126.go:24:2: WaitReasonSemacquire redeclared in this block
        pkg/sync/runtime_constants_go125.go:24:2: other declaration of WaitReasonSemacquire
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.18, NixOS, 26.05 (Yarara), 26.05.20260318.b40629e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/ap9dpkyzikzzh04259wlsvha2mw455x4-source`

### Notify maintainers

@gpl 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gv/gvisor/fix-go-mod-tidy.diff
+++ b/pkgs/by-name/gv/gvisor/fix-go-mod-tidy.diff
@@ -1,0 +1,40 @@
+diff --git a/go.mod b/go.mod
+index 60f4e32..87a9269 100644
+--- a/go.mod
++++ b/go.mod
+@@ -4,7 +4,6 @@ go 1.25.5
+ 
+ require (
+ 	github.com/BurntSushi/toml v1.4.1-0.20240526193622-a339e1f7089c
+-	github.com/bazelbuild/rules_go v0.44.2
+ 	github.com/cenkalti/backoff v2.2.1+incompatible
+ 	github.com/cilium/ebpf v0.12.3
+ 	github.com/containerd/cgroups v1.0.4
+@@ -44,7 +43,6 @@ require (
+ 	github.com/Microsoft/go-winio v0.6.0 // indirect
+ 	github.com/Microsoft/hcsshim v0.9.12 // indirect
+ 	github.com/containerd/ttrpc v1.1.2 // indirect
+-	github.com/creack/pty v1.1.24 // indirect
+ 	github.com/davecgh/go-spew v1.1.1 // indirect
+ 	github.com/docker/go-events v0.0.0-20190806004212-e31b211e4f1c // indirect
+ 	github.com/docker/go-units v0.4.0 // indirect
+@@ -53,7 +51,6 @@ require (
+ 	github.com/golang/protobuf v1.5.4 // indirect
+ 	github.com/google/go-cmp v0.7.0 // indirect
+ 	github.com/google/gofuzz v1.2.0 // indirect
+-	github.com/googleapis/enterprise-certificate-proxy v0.3.6 // indirect
+ 	github.com/googleapis/gnostic v0.5.5 // indirect
+ 	github.com/hashicorp/errwrap v1.1.0 // indirect
+ 	github.com/hashicorp/go-multierror v1.1.1 // indirect
+@@ -68,11 +65,9 @@ require (
+ 	go.opencensus.io v0.24.0 // indirect
+ 	golang.org/x/net v0.44.0 // indirect
+ 	golang.org/x/oauth2 v0.30.0 // indirect
+-	golang.org/x/telemetry v0.0.0-20250908211612-aef8a434d053 // indirect
+ 	golang.org/x/term v0.35.0 // indirect
+ 	golang.org/x/text v0.29.0 // indirect
+ 	golang.org/x/xerrors v0.0.0-20220907171357-04be3eba64a2 // indirect
+-	google.golang.org/api v0.249.0 // indirect
+ 	google.golang.org/genproto/googleapis/rpc v0.0.0-20250818200422-3122310a409c // indirect
+ 	google.golang.org/grpc v1.75.1 // indirect
+ 	gopkg.in/inf.v0 v0.9.1 // indirect

--- a/pkgs/by-name/gv/gvisor/package.nix
+++ b/pkgs/by-name/gv/gvisor/package.nix
@@ -12,7 +12,7 @@
 
 buildGoModule {
   pname = "gvisor";
-  version = "20251110.0";
+  version = "20260316.0";
 
   # gvisor provides a synthetic go branch (https://github.com/google/gvisor/tree/go)
   # that can be used to build gvisor without bazel.
@@ -21,8 +21,8 @@ buildGoModule {
   src = fetchFromGitHub {
     owner = "google";
     repo = "gvisor";
-    rev = "2617196c08506a30764bf6261b79d52797916dda";
-    hash = "sha256-qx1uCRTJVotSbTojBf/Nj8LfLdUvsnxMkPuyJjLLadM=";
+    rev = "7644cf3a343c39df4706458b27fa87e021187ac9";
+    hash = "sha256-L+SpxYthtuypW+6iP1wp7bqF8LiCwVNnzH7Obzh24ps=";
   };
 
   # Replace the placeholder with the actual path to ldconfig
@@ -31,7 +31,7 @@ buildGoModule {
       --replace-fail '"/sbin/ldconfig"' '"${glibc}/bin/ldconfig"'
   '';
 
-  vendorHash = "sha256-Ey4M3NK/+AVkr7r0aA+kAfNk1yVfnDn3Izy7u74HFkE=";
+  vendorHash = "sha256-8Zkgt5hegYEHnG1lF+wLgdru6t3l+Z/qKRvJHukZbPo=";
 
   nativeBuildInputs = [ makeWrapper ];
 
@@ -59,6 +59,8 @@ buildGoModule {
       }
     mv $out/bin/shim $out/bin/containerd-shim-runsc-v1
   '';
+
+  patches = [ ./fix-go-mod-tidy.diff ];
 
   passthru.tests = { inherit (nixosTests) gvisor; };
 

--- a/pkgs/by-name/gv/gvisor/package.nix
+++ b/pkgs/by-name/gv/gvisor/package.nix
@@ -12,7 +12,7 @@
 
 buildGoModule {
   pname = "gvisor";
-  version = "20260316.0";
+  version = "20260406.0";
 
   # gvisor provides a synthetic go branch (https://github.com/google/gvisor/tree/go)
   # that can be used to build gvisor without bazel.
@@ -21,8 +21,8 @@ buildGoModule {
   src = fetchFromGitHub {
     owner = "google";
     repo = "gvisor";
-    rev = "7644cf3a343c39df4706458b27fa87e021187ac9";
-    hash = "sha256-L+SpxYthtuypW+6iP1wp7bqF8LiCwVNnzH7Obzh24ps=";
+    rev = "db8d2c9abca39156c61ee2769d52b8a11accbe16";
+    hash = "sha256-T0ilLqZTX2KNZdR7wuMnYimnL+G5Tbkd77IULCZE764=";
   };
 
   # Replace the placeholder with the actual path to ldconfig

